File: app/src/main/java/com/morrowbone/mafiacards/app/application/MafiaApp.java
package com.morrowbone.mafiacards.app.application;

import android.content.Context;
import android.content.SharedPreferences;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import androidx.multidex.MultiDexApplication;

import com.google.android.gms.analytics.GoogleAnalytics;
import com.google.android.gms.analytics.Logger;
import com.google.android.gms.analytics.Tracker;
import com.morrowbone.mafiacards.app.R;

import java.util.HashMap;

public class MafiaApp extends MultiDexApplication {
    private volatile static MafiaApp mInstance;

    public MafiaApp() {
        super();
    }

    public static MafiaApp getInstance() {
        return mInstance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        mInstance = this;
    }
}

File: app/src/main/java/com/morrowbone/mafiacards/app/application/MafiaApp.kt
package com.morrowbone.mafiacards.app.application

import android.content.ContextWrapper
import androidx.multidex.MultiDexApplication
import com.morrowbone.mafiacards.app.data.AppDatabase
import com.morrowbone.mafiacards.app.data.DefaultDeckDao
import com.pixplicity.easyprefs.library.Prefs
import kotlinx.coroutines.Dispatchers.Default
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch

class MafiaApp : MultiDexApplication() {

    override fun onCreate() {
        super.onCreate()
        instance = this
        Prefs.Builder()
                .setContext(this)
                .setMode(ContextWrapper.MODE_PRIVATE)
                .setPrefsName(packageName)
                .setUseDefaultSharedPreference(true)
                .build()
        // I use this to force database updating
        val dataBaseVersion = AppDatabase.getInstance(this).openHelper.readableDatabase.version
        GlobalScope.launch(Default) {
            DefaultDeckDao.init()
        }
    }

    companion object {
        var instance: MafiaApp? = null
            private set
    }
}
